### ISSUE TITLE
HomeBridge won't start on Raspberry Pi

### ISSUE THREAD

**Issue by RaspJ** [nfarina/homebridge#2063]
I’m having problems getting HomeBridge to run on my Raspberry Pi and I’m getting desperate after hours and hours of trying to get it running. The thing is that I tried it on my MacBook and it works great, same settings / config file.

General info:
Raspberry Pi 3B+
OS: Raspbian GNU/Linux 9 (stretch)
node version: v8.11.1
npm version: 6.4.1 (updated after the installation of HomeBridge though)
HomeBridge version: 0.4.45
I have the libavahi-compat-libdnssd-dev installed

Here’s the information from terminal when I’m trying to start HomeBridge:

pi@raspberrypi:~ $ homebridge 
Loaded config.json with 0 accessories and 1 platforms.
---
====================
ERROR LOADING PLUGIN homebridge-telldus:
Error: Cannot find module 'debug'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge-telldus/index.js:4:15)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Plugin.load (/usr/local/lib/node_modules/homebridge/lib/plugin.js:72:23)
    at Server.<anonymous> (/usr/local/lib/node_modules/homebridge/lib/server.js:149:14)
    at Array.forEach (<anonymous>)
    at Server._loadPlugins (/usr/local/lib/node_modules/homebridge/lib/server.js:141:22)
====================
No plugins found. See the README for information on installing plugins.
Loading 1 platforms...
/usr/local/lib/node_modules/homebridge/lib/api.js:122
      throw new Error("The requested platform '" + name + "' was not registered by any plugin.");
      ^

Error: The requested platform 'Telldus' was not registered by any plugin.
    at API.platform (/usr/local/lib/node_modules/homebridge/lib/api.js:122:13)
    at Server._loadPlatforms (/usr/local/lib/node_modules/homebridge/lib/server.js:316:45)
    at Server.run (/usr/local/lib/node_modules/homebridge/lib/server.js:86:36)
    at module.exports (/usr/local/lib/node_modules/homebridge/lib/cli.js:45:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3

This is my config.json located in ~/.homebridge/

{
	    "bridge": {
	        "name": "Homebridge",
	        "username": "CC:22:3D:E3:CE:30",
	        "port": 51826,
	        "pin": "031-45-154"
	    },
	    
	    "description": "This is an example configuration file with one fake accessory and one fake platform. You can use this as a template for creating your own configuration file containing devices you actually own.",
	

	    "accessories": [
	    ],
	

	    "platforms": [
	        {
	            "platform" : "Telldus",
	            "name" : "Telldus Live",
	            "public_key" : "myPublicKey",
	            "private_key" : "myPrivateKey",
	            "token" : "myToken",
	            "token_secret" : "mySecretToken"
	        }
	    ]
	}

Any ideas? For example, what does " Error: Cannot find module 'debug' " mean?!

**Comment by Frank-Friemel** [comment]
Please try _npm install -g debug_

**Comment by RaspJ** [comment]
This is what happened:

sudo npm install -g debug
+ debug@4.1.0
added 2 packages from 3 contributors in 15.131s
homebridge
Loaded config.json with 0 accessories and 1 platforms.
---
====================
ERROR LOADING PLUGIN homebridge-telldus:
Error: Cannot find module 'debug'

Then still the same error as before...

I wanted to reinstall HomeBridge and homebridge-telldus but I can't uninstall them. I've tried sudo npm uninstall -g homebridge and the same thing with homebridge-telldus. Even without sudo, but homebridge is still there.

Can the problem be that something is in the wrong location? The homebridge-telldus folder or something about 'debug' that I have to mv? Since homebridge is unable to find any plugins I mean. But then again... it didn't start without the telldus addon either.

**Comment by RaspJ** [comment]
Progress but still not there...

Same error when I start with the basic command: homebridge. But if i start with this command when I'm in the /usr/lib/node_modules/homebridge/ it starts: ./bin/homebridge

Everything looks great, homebridge is up and running and it finds all my stuff from Telldus too, but now I can't add it in Home on my iPhone. It doesn't show up. I've tried to disable ipv6 because I read in other issues that this was often the problem but still no luck. I tried removing the persist folder too. My Raspberry Pi is connected both with ethernet cable and wifi. Same wifi as my phone. Does anyone have a clue what to do now?
